pyvrp/tests/test_ProblemData.py

```diff
@@ -83,6 +83,96 @@ def test_raises_for_invalid_client_data(
         Client(x, y, demand, service, tw_early, tw_late, release_time, prize)
 
 
+@mark.parametrize(
+    "x,y,demand,service,tw_early,tw_late,release_time,prize",
+    [
+        (0, 0, 1, 0, 0, 0, 0, 0),  # demand != 0
+        (0, 0, 0, 1, 0, 0, 0, 0),  # service duration != 0
+        (0, 0, 0, 0, 0, 0, 1, 0),  # release time != 0
+    ],
+)
+def test_raises_for_invalid_depot_data(
+    x: int,
+    y: int,
+    demand: int,
+    service: int,
+    tw_early: int,
+    tw_late: int,
+    release_time: int,
+    prize: int,
+):
+    depot = Client(
+        x, y, demand, service, tw_early, tw_late, release_time, prize
+    )
+
+    with assert_raises(ValueError):
+        ProblemData([depot], [VehicleType(1, 2)], [[0]], [[0]])
+
+
+def test_problem_data_raises_when_no_clients_provided():
+    """
+    Tests that the ``ProblemData`` constructor raises a ``ValueError`` when
+    no clients are provided.
+    """
+    with assert_raises(ValueError):
+        ProblemData(
+            clients=[],
+            vehicle_types=[VehicleType(1, 2)],
+            distance_matrix=[],
+            duration_matrix=[],
+        )
+
+    # One client (the depot) should not raise.
+    ProblemData(
+        clients=[Client(x=0, y=0)],
+        vehicle_types=[VehicleType(1, 2)],
+        distance_matrix=[[0]],
+        duration_matrix=[[0]],
+    )
+
+
+def test_problem_data_matrix_dimensions():
+    """
+    Tests that the ``ProblemData`` constructor raises a ``ValueError`` when
+    the distance or duration matrix has less rows or columns than the number
+    of clients. The number of rows and columns are allowed to be larger than
+    the number of clients.
```

Comment:
I'm not sure if we should allow this. Why would we not want to enforce that `mat.shape = (dimension, dimension)`?

Reply:
Yeah, I changed this. Not sure why I think I thought it was a good idea to pass "larger" matrices.